packages/sheets-numfmt/src/commands/commands/set-currency.command.ts

```diff
@@ -28,14 +29,17 @@ export const SetCurrencyCommand: ICommand = {
     handler: async (accessor: IAccessor) => {
         const commandService = accessor.get(ICommandService);
         const selectionManagerService = accessor.get(SheetsSelectionsService);
+        const localeService = accessor.get(LocaleService);
+        const locale = localeService.getCurrentLocale();
```

Comment:
Is this going to work for euros ? 
Or will we have to wait until univer is translated into every european language ?
I liked using the browser's current regional settings, because it allowed this to work with languages univer is not (and will never be) translated into.

Reply:
No, no, no. The European region is also a future direction for our services. It's just that we currently don't have enough manpower to support this work. Most of the internationalization resources in the project are provided by the community, and we are very grateful for their help.
Can we try adding an enumeration of LocaleType euro, but translate the text in English (Wait for the open source community to provide the corresponding language copywriting)?

Reply:
If you want to implement this button in the toolbar by yourself, you can try overriding this menu configuration.
After loading the numfmt-plugin, you can reset the(CurrencyMenuItem)[packages/sheets-numfmt/src/menu/menu.ts] by calling the(setMenuItem)[packages/ui/src/services/menu/menu.service.ts] method in menu.service.ts.

Reply:
The thing is, there are many many different languages in the "euro region". What would an euro locale contain?

Reply:
Are you sure you don't want to decouple supported languages from supported currencies?

Reply:
We haven't thought about this for a while, but from a design point of view,
the current language isn't platform-specific, it's available through the LocaleService in project

Reply:
【there are many many different languages in the "euro region"】Could there be multiple languages for the same currency? That means that our localType enumerations may be excessive.
Please give me a little more time to discuss it internally

Reply:
Ideally, I would like the default currency displayed to SQLPage users to be the currency of the place they are in, even if the rest of the UI is in English. 

There are 20 countries in the eurozone, speaking 19 languages. Some of these languages have <1M locutors. I understand you have grand ambitions, but univerjs will probably not be translated in all of these languages anytime soon. It would be great if we could still have the € sign by default...

Reply:
Sure, I think providing an API to set a custom currency symbol is a great solution. It allows you to set the currency symbol independently of the platform language and `Univer` own language settings. When integrating with `Univer`, you can use the API to set the currency symbol, regardless of how you obtain the system language.

Reply:
I think there is a misunderstanding. I don't need an api to set the default currency symbol, I would like the default symbol to automatically reflect the users preference, wherever they are. That's what my initial pull request did.

Reply:
1. We will provide an API for setting the currency symbol, and you can write some code to call it. For example, you can use the browser's user-agent to recognize the country code and set the corresponding symbol.

2. We will add a country code identifier to the snapshot upon creation. If not set, it will default to the mapped value of that country code and will not follow the language settings.
 
3. Later, we will provide a StackBlitz code to guide you through the implementation.

There are some blocking bugs that need to be resolved, and we expect to provide them to you by next week.